auditing_standards.md

```diff
@@ -237,3 +237,76 @@ Crates with this criteria contain very dangerous unsafe rust code.
 *   Most crates that try to be sound but don't quite make the cut go in `ub-risk-3`, not here. These crates are wildly unsafe, and the only time we should use them is when they are a necessary evil.
 *   Everything worse than `ub-risk-3` goes in here and we should do our best to avoid using them.
+
+## Criteria-agnostic guidelines
+
+### Delta audits should describe the final version
+
+In general, delta audits should always take the previous, baseline into
+account.  This means that inspecting just the source code delta should never result
+in lowering the `UB-risk-N` value from the previous audit, or downgrading `crypto-safe` to
+`does-not-contain-crypto`.  OTOH delta audits can lower the UB risk if the delta
+fixes all the issues that were identified in the previous, baseline audit.
+
+For example, even though
+[the 1.0.79 to 1.0.80 delta of the `proc-macro2` crate](https://chromium-review.googlesource.com/c/chromium/src/+/5453972/2..6)
+doesn't add or change any `unsafe` code, it shouldn't be recorded as
+`ub-risk-0`, because there is still `unsafe` outside of the delta
+(e.g. `unsafe { imp::Literal::from_str_unchecked(repr) }` in
+[`lib.rs` on line 1282](https://chromium-review.googlesource.com/c/chromium/src/+/5453972/2..6/third_party/rust/chromium_crates_io/vendor/proc-macro2-1.0.80/src/lib.rs#1282)).
+
+### Audits should ignore criteria required by `config.toml` policy
+
+Audits should record the actual risk of the crate, even if a looser
+certification would be sufficient to pass a `config.toml` policy.
+
+For example, Chromium requires `ub-risk-2` for crates used in the browser
+process - e.g. see the following auto-generated `config.toml`
+[entry for the `quote` crate](https://source.chromium.org/chromium/chromium/src/+/main:third_party/rust/chromium_crates_io/supply-chain/config.toml;l=158-159;drc=aaf3f1dab40ed4248391227583a1590bfbf7062e):
+
+```
+# supply-chain/config.toml:
+[policy."quote:1.0.36"]
+criteria = ["does-not-implement-crypto", "safe-to-deploy", "ub-risk-2"]
+```
+
+This means that certifying this crate as `ub-risk-2` is sufficient for `cargo
+vet` checks/presubmits to pass and `cargo vet check` will actually only ask
+to certify at the `ub-risk-2` level - see the example below:
+
+```
+$ tools/crates/run_cargo_vet.py check
+Vetting Failed!
+
+1 unvetted dependencies:
+  quote:1.0.36 missing ["safe-to-deploy", "does-not-implement-crypto", "ub-risk-2"]
+
+recommended audits for safe-to-deploy, does-not-implement-crypto, ub-risk-2:
+    Command                             Publisher  Used By                                    Audit Size
+    cargo vet diff quote 1.0.35 1.0.36  dtolnay    syn, chromium, prost-derive, and 6 others  6 files changed, 568 insertions(+), 548 deletions(-)
+
+estimated audit backlog: 1116 lines
+
+Use |cargo vet certify| to record the audits.
+```
+
+Nevertheless, the audit should ignore the policy requirements and certify
+`quote` 1.0.36 as `ub-risk-0` because it doesn't have any `unsafe` code.
```

Comment:
```suggestion
Although it is recommended to audit at `ub-risk-2`, the audit should always
certify `quote` 1.0.36 at the most precise ub-risk level possible. In this case,
the crate should be audited as `ub-risk-0` because it doesn't contain any
unsafe code.
```